pkgs/top-level/all-packages.nix

```diff
@@ -4161,8 +4161,12 @@ let
   erlangR16_odbc = callPackage ../development/interpreters/erlang/R16.nix { odbcSupport = true; };
   erlangR17 = callPackage ../development/interpreters/erlang/R17.nix { };
   erlangR17_odbc = callPackage ../development/interpreters/erlang/R17.nix { odbcSupport = true; };
+  erlangR17_javac = callPackage ../development/interpreters/erlang/R17.nix { javacSupport = true; };
+  erlangR17_odbc_javac = callPackage ../development/interpreters/erlang/R17.nix { javacSupport = true; odbcSupport = true; };
```

Comment:
Understood, I considered that, but the all or nothing approach slightly annoys me. Honestly I think this will appease Erlang programmers. Really the 2 main options as far as my experience tells is java and odbc. Why? Deploying a bloated useless Java to production when you simply wanted odbc just grates me. So unless there is an over riding *rough* consensus on an all or nothing strategy I'm going to stick to my guns. Having said that I'm more than happy to make the change. Lets hear what others have to say :-) As for Hipe, that should be mandatory.